astropy/utils/data.py

```diff
@@ -894,21 +914,40 @@ def _find_pkg_data_path(data_name, package=None):
         else:
             package = module.__package__
     else:
+        # package errors if it isn't a str
+        # so there is no need for checks in the containing if/else
         module = resolve_name(package)
 
-    rootpkgname = package.partition('.')[0]
-
-    rootpkg = resolve_name(rootpkgname)
-
+    # module path within package
     module_path = os.path.dirname(module.__file__)
-    path = os.path.join(module_path, data_name)
+    full_path = os.path.join(module_path, *path)
 
+    # Check that file is inside tree.
+    rootpkgname = package.partition('.')[0]
+    rootpkg = resolve_name(rootpkgname)
     root_dir = os.path.dirname(rootpkg.__file__)
-    if not _is_inside(path, root_dir):
+    if not _is_inside(full_path, root_dir):
         raise RuntimeError(f"attempted to get a local data file outside "
                            f"of the {rootpkgname} tree.")
 
-    return path
+    return full_path
+
+
+@deprecated('4.3', alternative='Use astropy.utils.data.get_pkg_data_path',
```

Comment:
@embray , why deprecate a hidden function? The policy is that we don't have to if it is hidden. People use it at their own risk.

Reply:
@embray , just FYI that `pending=True` is basically a no-op. What is your plan going forward with this deprecation?

Also, I still don't think we have to deprecate a hidden function.